test/test_backports.py

```diff
@@ -0,0 +1,51 @@
+"""Tests for :mod:`qmflows.backports`."""
+
+import sys
+from typing import Type, Union
+
+from assertionlib import assertion
+
+from qmflows.backports import (Literal, Final, nullcontext,
+                               _LiteralBackup, _FinalBackup, _NullContextBackup)
+
+
+def test_literal():
+    """Tests for :data:`Literal` and :data:`_LiteralBackup`."""
+    if sys.version_info.minor >= 8:
+        assertion.eq(Literal.__module__, 'typing')
+    else:
+        assertion.eq(Literal.__module__, 'typing_extensions')
+
+    literal = _LiteralBackup()
```

Comment:
Could you explain to me what `_LiteralBackup` doing?

Reply:
It's creates an instance whose ``__getitem__()`` method has the same signature as ``typing.Literal`` (which was only added in Python 3.8).
As properly backporting ``typing.Literal`` is kinda difficult, it just converts the passed name into the appropriate object type.

```
from typing import Union, Type

Literal = _LiteralBackup()

Literal[1] == int
Literal[int] == Type[int]
Literal[1, 'two', 3.0] == Union[int, str, float]
```